### PR TITLE
[FIX] account_edi_ubl_cii: PartyTaxScheme should begin with country code

### PR DESCRIPTION
**Pull request by JulienVR**
In Peppol Bis 3, the value on PartyTaxScheme/CompanyID should be prefixed with a 2 letters code representing the country of the partner. Currently, filling a VAT without this country code will lead to the error: "[BR-CO-09]-The Seller VAT identifier (BT-31), the Seller tax representative VAT identifier (BT-63) and the Buyer VAT identifier (BT-48) shall have a prefix in accordance with ISO code ISO 3166-1 alpha-2 by which the country of issue may be identified. Nevertheless, Greece may use the prefix ‘EL’."

On the other hand, some extra-european countries use Peppol Bis 3 but do not prepend their VAT with the country code (i.e. Australia). To allow such cases, we automatically add the country code when it's missing.

To prevent regression, we also add a contraint mimicking the schematron: we check that the final value of the PartyTaxScheme/CompanyID is prefixed with a 2 letters code. This way, the constraint is independant from the code.

### CODE REVIEW

File: addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py [hunk 1]
```diff
@@ -57,10 +57,13 @@ def _get_partner_party_tax_scheme_vals_list(self, partner, role):
...
-            # /!\ For Australian companies, the ABN is encoded on the VAT field, but doesn't have the 2 digits prefix,
-            # causing a validation error
-            if partner.country_id.code == "AU" and partner.vat and not partner.vat.upper().startswith("AU"):
-                vals['company_id'] = "AU" + partner.vat
+            # Some extra european countries use Bis 3 but do not prepend their VAT with the country code (i.e.
+            # Australia). Allow them to use Bis 3 without raising BR-CO-09.
+            if partner.country_id \
+                    and partner.country_id not in self.env.ref('base.europe').country_ids \
+                    and len(partner.vat) >= 2 \
+                    and not partner.vat[:2].isalpha():
```
Comment (smetl) [inline addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py:65]:
```suggestion
            if (
                partner.country_id
                and partner.country_id not in self.env.ref('base.europe').country_ids
                and len(partner.vat) >= 2
                and not partner.vat[:2].isalpha()
            ):
```

File: addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py [hunk 2]
```diff
@@ -369,6 +374,18 @@ def _invoice_constraints_cen_en16931_ubl(self, invoice, vals):
                 # /!\ exception: possible to have any number of ecotaxes (fixed tax) with a regular percentage tax
                 constraints.update({'cen_en16931_tax_line': _("Each invoice line shall have one and only one tax.")})
 
+        for role in ('supplier', 'customer'):
+            constraints[f'cen_en16931_{role}_country'] = self._check_required_fields(vals[role], 'country_id')
+            scheme_vals = vals['vals'][f'accounting_{role}_party_vals']['party_vals']['party_tax_scheme_vals'][-1:]
+            if not (scheme_vals and scheme_vals[0]['company_id'] and scheme_vals[0]['company_id'][:2].isalpha()) \
+                and (scheme_vals and scheme_vals[0]['tax_scheme_id'] == 'VAT') \
+                and self._name in ('account.edi.xml.ubl_bis3', 'account.edi.xml.ubl_nl', 'account.edi.xml.ubl_de'):
```
Comment (smetl) [inline addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py:382]:
```suggestion
if (
    not (scheme_vals and scheme_vals[0]['company_id'] and scheme_vals[0]['company_id'][:2].isalpha())
    and (scheme_vals and scheme_vals[0]['tax_scheme_id'] == 'VAT')
    and self._name in ('account.edi.xml.ubl_bis3', 'account.edi.xml.ubl_nl', 'account.edi.xml.ubl_de')
):
```
sorry but I can't read that part without vomiting :vomiting_face:

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by JulienVR** [comment]
@robodoo rebase-ff

**Comment by smetl** [comment]
@robodoo rebase-ff

**Pull request closed without merging**